docs/reference/how-to/knn-search.asciidoc

```diff
@@ -0,0 +1,98 @@
+[[tune-knn-search]]
+== Tune approximate kNN search
+
+{es} supports <<approximate-knn, approximate k-nearest neighbor search>> for
+efficiently finding the _k_ nearest vectors to a query vector. Since
+approximate kNN search works differently from other queries, there are special
+considerations around its performance.
+
+Many of these recommendations help improve search speed. With approximate kNN,
+the indexing algorithm runs searches under the hood to create the vector index
+structures. So these same recommendations also help with indexing speed.
+
+[discrete]
+=== Ensure data nodes have enough memory
+
+{es} currently uses the https://arxiv.org/abs/1603.09320[HNSW] algorithm for
+approximate kNN search. HNSW is a graph-based algorithm which only works
+efficiently when all vector data is held in memory. You should ensure that
+data nodes have enough RAM to hold the vector data and index structures. To
+check the size of the vector data, you can use the <<indices-disk-usage>> API.
+The RAM that's required is for the filesystem cache, which is separate from
+the Java heap.
+
+include::search-speed.asciidoc[tag=warm-fs-cache]
+
+[discrete]
+=== Reduce vector dimensionality
+
+The speed of kNN search scales linearly with the number of vector dimensions,
+because each similarity computation considers each element in the two vectors.
+Whenever possible, it's better to use vectors with a lower dimension. Some
+embedding models come in different "sizes", with both lower and higher
+dimensional options available. You could also experiment with dimensionality
+reduction techniques like PCA. When experimenting with different approaches,
+it's important to measure the impact on relevance to ensure the search
+quality is still acceptable.
+
+[discrete]
+=== Prefer `dot_product` over `cosine`
+
+When indexing vectors for approximate kNN search, you need to specify the
+<<dense-vector-similarity, `similarity` function>> for comparing the vectors.
+If you'd like to compare vectors through cosine similarity, there are two
+options.
+
+The `cosine` option accepts any float vector and computes the cosine
+similarity. While this is convenient for testing, it's not the most efficient
+approach. Instead, we recommend using the `dot_product` option for cosine
```

Comment:
did you mean here "for vector similarity" instead of "cosine similarity"?

Reply:
I did mean "cosine similarity", because `dot_product` is being used to compute the cosine similarity between the two vectors. I see how this is confusing though, I could just say "Instead, we recommend using the `dot_product` option."